src/WebJobs.Extensions/Extensions/Timers/Scheduling/ScheduleMonitor.cs

```diff
@@ -56,32 +56,57 @@ public abstract class ScheduleMonitor
                 lastStatus = new ScheduleStatus
                 {
                     Last = default(DateTime),
-                    Next = nextOccurrence
+                    Next = nextOccurrence,
+                    LastUpdated = now
                 };
                 await UpdateStatusAsync(timerName, lastStatus);
                 recordedNextOccurrence = nextOccurrence;
             }
             else
             {
                 DateTime expectedNextOccurrence;
-                if (lastStatus.Last == default(DateTime))
+
+                // Track the time that was used to create 'expectedNextOccurrence'.
+                DateTime lastUpdated;
+
+                if (lastStatus.LastUpdated != default(DateTime))
                 {
-                    // there have been no executions of the function yet, so compute
-                    // from now
-                    expectedNextOccurrence = schedule.GetNextOccurrence(now);
+                    // If we know when we last calculated the stored 'Next', use the current schedule
+                    // to recalculate. It may have changed, which will be checked below.
+                    expectedNextOccurrence = schedule.GetNextOccurrence(lastStatus.LastUpdated);
+                    lastUpdated = lastStatus.LastUpdated;
                 }
-                else
+                else if (lastStatus.Last != default(DateTime))
```

Comment:
If LastUpdated exists, it will always be the same as Last. I'll look and see if I can easily swap them around and still have it be readable (I think I can).